### PR TITLE
Add types for `.distinctOn`

### PR DESCRIPTION
**Pull request by victorandree**
Interface as provided by #3513.

There's a separate PR for docs at https://github.com/knex/documentation/pull/257

### CODE REVIEW

File: types/index.d.ts [hunk 1]
```diff
@@ -907,6 +908,16 @@ declare namespace Knex {
...
 
+  interface DistinctOn<TRecord extends {}, TResult = {}[]> {
+    <ColNameUT extends keyof TRecord>(
+      ...columnNames: readonly ColNameUT[]
+    ): QueryBuilder<TRecord, TResult>;
+
+    <ColNameUT extends keyof TRecord>(
+      columnNames: readonly ColNameUT[]
+    ): QueryBuilder<TRecord, TResult>;
+  }
```
Comment (lorefnon) [inline types/index.d.ts:919]: It is also good to add another signature with `columnNames: readonly string[]` to accomodate things like `distinctOn('users.foo')` where the argument is not the same as column key.

**Comment by lorefnon** [comment]
@victorandree Thanks for your PR. Besides the minor change requested, this looks good.

**Comment by victorandree** [comment]
Thanks @lorefnon! I force pushed to add 2 string interfaces:

```typescript
(...columnNames: readonly string[]): QueryBuilder<TRecord, TResult>;
(columnNames: readonly string[]): QueryBuilder<TRecord, TResult>;
```

**Comment by kibertoad** [comment]
Released in 0.20.15

**Pull request merged**